src/V3EmitXml.cpp

```diff
@@ -114,6 +114,7 @@ class EmitXmlFileVisitor : public AstNVisitor {
         puts(" origName="); putsQuoted(nodep->origName());
         if (nodep->level()==1 || nodep->level()==2)  // ==2 because we don't add wrapper when in XML mode
             puts(" topModule=\"1\"");  // IEEE vpiTopModule
+        if (nodep->modPublic()) puts(" public=\"true\"");
```

Comment:
Suggest pull these xml improvements into a separate pull request. Once regressions pass you can squash and push to master.

Reply:
Thanks, see #2079